backbone.js

```diff
@@ -1487,11 +1483,12 @@
       this._wantsHashChange = this.options.hashChange !== false;
       this._hasHashChange   = 'onhashchange' in window;
       this._wantsPushState  = !!this.options.pushState;
-      this._hasPushState    = !!(this.options.pushState && this.history && this.history.pushState);
+      this._hasPushState    = !!(this.history && this.history.pushState);
+      this._usePushState    = this._wantsPushState && this._hasPushState;
```

Comment:
Regardless of where `root` ends up, I like this. I've always thought the way we conflate `_wantsPushState` and `_hasPushState` will eventually cause a bug. :clap: